diff --git a/cpp/src/umap/fuzzy_simpl_set/naive.cuh b/cpp/src/umap/fuzzy_simpl_set/naive.cuh
index 2462931c6a..c8e27ce345 100644
--- a/cpp/src/umap/fuzzy_simpl_set/naive.cuh
+++ b/cpp/src/umap/fuzzy_simpl_set/naive.cuh
@@ -315,6 +315,7 @@ void launcher(nnz_t n,
 
   raft::sparse::COO<value_t> in(stream, n * n_neighbors, n, n);
 
+  /*
   // check for logging in order to avoid the potentially costly `arr2Str` call!
   if (ML::default_logger().should_log(rapids_logger::level_enum::debug)) {
     CUML_LOG_DEBUG("Smooth kNN Distances");
@@ -323,6 +324,7 @@ void launcher(nnz_t n,
     str = raft::arr2Str(rhos.data(), 25, "rhos", stream);
     CUML_LOG_DEBUG("%s", str.c_str());
   }
+  */
 
   RAFT_CUDA_TRY(cudaPeekAtLastError());
 
@@ -346,12 +348,14 @@ void launcher(nnz_t n,
                                        to_process);
   RAFT_CUDA_TRY(cudaPeekAtLastError());
 
+  /*
   if (ML::default_logger().should_log(rapids_logger::level_enum::debug)) {
     CUML_LOG_DEBUG("Compute Membership Strength");
     std::stringstream ss;
     ss << in;
     CUML_LOG_DEBUG(ss.str().c_str());
   }
+  */
 
   /**
    * Combines all the fuzzy simplicial sets into a global
diff --git a/cpp/src/umap/simpl_set_embed/algo.cuh b/cpp/src/umap/simpl_set_embed/algo.cuh
index 1e10691ab6..d6b2501792 100644
--- a/cpp/src/umap/simpl_set_embed/algo.cuh
+++ b/cpp/src/umap/simpl_set_embed/algo.cuh
@@ -339,11 +339,13 @@ void launcher(
 
   make_epochs_per_sample(out.vals(), out.nnz, n_epochs, epochs_per_sample.data(), stream);
 
+  /*
   if (ML::default_logger().should_log(rapids_logger::level_enum::debug)) {
     std::stringstream ss;
     ss << raft::arr2Str(epochs_per_sample.data(), out.nnz, "epochs_per_sample", stream);
     CUML_LOG_DEBUG(ss.str().c_str());
   }
+  */
 
   optimize_layout<T, nnz_t, TPB_X>(embedding,
                                    m,
diff --git a/cpp/src/umap/supervised.cuh b/cpp/src/umap/supervised.cuh
index 0b2682bfff..997e9a6a37 100644
--- a/cpp/src/umap/supervised.cuh
+++ b/cpp/src/umap/supervised.cuh
@@ -301,6 +301,7 @@ void perform_general_intersection(const raft::handle_t& handle,
     handle, y_inputs, y_inputs, knn_graph, params->target_n_neighbors, params, stream);
   RAFT_CUDA_TRY(cudaPeekAtLastError());
 
+  /*
   if (ML::default_logger().should_log(rapids_logger::level_enum::debug)) {
     CUML_LOG_DEBUG("Target kNN Graph");
     std::stringstream ss1, ss2;
@@ -311,6 +312,7 @@ void perform_general_intersection(const raft::handle_t& handle,
       y_knn_dists.data(), rgraph_coo->n_rows * params->target_n_neighbors, "knn_dists", stream);
     CUML_LOG_DEBUG("%s", ss2.str().c_str());
   }
+  */
 
   /**
    * Compute fuzzy simplicial set
@@ -326,12 +328,14 @@ void perform_general_intersection(const raft::handle_t& handle,
                                                        stream);
   RAFT_CUDA_TRY(cudaPeekAtLastError());
 
+  /*
   if (ML::default_logger().should_log(rapids_logger::level_enum::debug)) {
     CUML_LOG_DEBUG("Target Fuzzy Simplicial Set");
     std::stringstream ss;
     ss << ygraph_coo;
     CUML_LOG_DEBUG(ss.str().c_str());
   }
+  */
 
   /**
    * Compute general simplicial set intersection.
