diff --git a/python/cuml/cuml/experimental/accel/_wrappers/sklearn.py b/python/cuml/cuml/experimental/accel/_wrappers/sklearn.py
index 9b7a09b887..d065e832f4 100644
--- a/python/cuml/cuml/experimental/accel/_wrappers/sklearn.py
+++ b/python/cuml/cuml/experimental/accel/_wrappers/sklearn.py
@@ -1,5 +1,5 @@
 #
-# Copyright (c) 2024, NVIDIA CORPORATION.
+# Copyright (c) 2024-2025, NVIDIA CORPORATION.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
@@ -127,3 +127,20 @@
     accelerated_module="cuml.neighbors",
     original_class_name="KNeighborsRegressor",
 )
+
+###############################################################################
+#                              Ensemble  Estimators                           #
+###############################################################################
+
+
+RandomForestRegressor = intercept(
+    original_module="sklearn.ensemble",
+    accelerated_module="cuml.ensemble",
+    original_class_name="RandomForestRegressor",
+)
+
+RandomForestClassifier = intercept(
+    original_module="sklearn.ensemble",
+    accelerated_module="cuml.ensemble",
+    original_class_name="RandomForestClassifier",
+)
